src/main/kotlin/no/nav/syfo/behandler/kafka/dialogmeldingfrombehandler/KafkaDialogmeldingFromBehandler.kt

```diff
@@ -40,11 +45,33 @@ fun pollAndProcessDialogmeldingFromBehandler(
     }
 }
 
-// TODO Hva skal denne faktisk gjøre?
 fun updateBehandlerOffice(
     consumerRecords: ConsumerRecords<String, KafkaDialogmeldingFromBehandlerDTO>,
 ) {
     consumerRecords.forEach {
-        log.info("Received dialogmelding record with key: ${it.key()}")
+        val dialogmeldingFromBehandler = it.value()
+        val partnerReferanse = getPartnerReferanse(dialogmeldingFromBehandler.fellesformatXML)
+        log.info("Received dialogmelding from partnerReferanse: $partnerReferanse")
     }
 }
+
+fun getPartnerReferanse(fellesformatXML: String): String {
+    val reader = StringReader(fellesformatXML)
+    val xmlInputFactory = XMLInputFactory.newFactory()
+    val xmlStreamReader = xmlInputFactory.createXMLStreamReader(reader)
+
+    while (xmlStreamReader.hasNext()) {
+        if (xmlStreamReader.isStartElement && xmlStreamReader.localName == "MottakenhetBlokk") {
+            break
+        }
+        xmlStreamReader.next()
+    }
+
+    val mottakenhetBlokk = mottakenhetBlokkUnmarshaller.unmarshal(xmlStreamReader) as XMLMottakenhetBlokk
+    xmlStreamReader.close()
```

Comment:
Burde man ha en try / finally block her som sørger for å alltid kjører .close på xmlStreamReader ?
Kan også bruke reader.use {} så vi sørger for at reader alltid lukkes